docs/X_layered_XCP.adoc

```diff
@@ -0,0 +1,109 @@
+
+=== Rough Outline of the Approach
+
+The XCP description file for the vECU wrapped as FMU will be placed into the `/extra/org.modelica.fmi.layered_XCP` folder.
+This description file follows the https://www.asam.net/standards/detail/mcd-2-mc/[ASAP MCD-2 MC] standard (aka `ASAP2`, also `A2L`) and customarily carries the file extension `.a2l`.
+
+[quote, ASAM e.V., https://www.asam.net/standards/detail/mcd-2-mc/[ASAP MCD-2 MC]]
+____
+ASAM MCD-2 MC (aka ASAP2) defines the description format of the internal ECU variables used in measurement and calibration.
+Measurement & calibration systems (MC-systems) require this description for both the parameterization of scalar constants, curves and maps of the ECU software and for recording the system's response via measurement variables during real-time testing.
+The description contains information about data types, dimensions, record layouts and memory locations of ECU variables.
+The standard also describes how the variable values are converted into human-readable quantities and displayed in an MC-system.
+____
+
+=== Comments about this Approach
+
+_[This standard will have no effect on the FMU interface, nor the C-API behavior._
+_The XCP server inside the FMU will most likely run in its own thread and leave the rest of the operation of the FMU unaffected._
+_The FMU must configure the XCP server and trigger certain actions and provide timing information._
+_This approach can also be used for plant models when the FMU exporter can generate an appropriate `A2L` file.]_
+
+== Details
+
+=== A2L Description Files
+
+FMI 3.0 introduces an `/extra` folder in its `zip` structure.
+The A2L description files will be placed under the reverse domain name controlled by the MAP FMI: `/extra/org.modelica.fmi.layered_XCP`.
+All versions of the `A2L` standard are allowed and it is the MCD tool's responsibility to handle each version correctly.
+It is currently not recommended to supply multiple versions of the `A2L` file due to their considerable size.
+The root name of the description files shall be identical to the FMU name and are case sensitive.
+
+This standard forbids the use of the `including mechanism` of additional `A2L` files to simplify complete extraction and copying of `A2L` files.
+
+It is recommended to include `IF_DATA XCP` elements to help MCD tools to connect and interact with the XCP server inside the FMU more reliably and without user interaction.
+
+=== Server Startup and Shutdown
+
+It is recommended to start the XCP service during `fmi3EnterInitializationMode` (or earlier) and shut down during `fmi3Terminate` (or later) if the FMU has no explicit power-up signal to simplify user interactions between simulator and MCD tool.
+
+If the FMU contains a virtual ECU with power-up control (K15), all build in OS and Basic Software services (including XCP) should follow normal power-up protocol.
+
+=== Variable Visibility
```

Comment:
Add a section about measurement rasters:
2.x Measurement Rasters
The A2L/XCP standards allow to measure variables synchronous to different time- or angular-based events. This requires to invoke the XCP service with a so-called raster ID in the thread which is executed for an event. Calibration is typically performed in a background thread. Refer to the A2L/XCP standards for more information.

Reply:
Added as non-normative hint.